src/elements/Link/Link.tsx

```diff
@@ -3,9 +3,22 @@ import { color as styledColor, space, SpaceProps } from "styled-system"
 import { color } from "../../helpers"
 import { Color } from "../../Theme"
 
+enum UnderlineBehavior {
+  Default = "default",
+  Hover = "hover",
+  None = "none",
+}
+
+const computeUnderline = (state: string, behavior: UnderlineBehavior): string => {
+  const blocklist: UnderlineBehavior[] = state === "hover" ? [UnderlineBehavior.None] : [UnderlineBehavior.Hover, UnderlineBehavior.None]
+  const none = blocklist.includes(behavior)
+  return none ? "none" : "underline"
+}
+
 export interface LinkProps extends SpaceProps {
-  noUnderline?: boolean
```

Comment:
Like @mzikherman mentioned, it would be best to maintain this prop for now while mapping old behavior to the new API. Then once that's out can grep over Reaction to see what currently uses `noUnderline` and replace with the new API. Then once that's done, can come back to Palette and open a new PR with this deprecated prop removed and mark as a MAJOR release, then manually update Reaction with the new Palette version (since Major updates don't automerge).